src/main/java/edu/rosehulman/covidtracer/service/PersonService.java

```diff
@@ -23,13 +23,20 @@
     private void showRepositoryInfo() {
         System.out.println("************PersonRepository autowired************  \n ->" + repository.getClass().getCanonicalName());
     }
-    
+
     public List<Person> getAllPersons(){
-        List<Person> result = repository.findAll();
-        if(!result.isEmpty()){
-            return result;
+        return repository.findAll();
+    }
+
+    public String getNameFromEmail(String email){
+        List<Person> people = repository.findByEmail(email);
+        System.out.println("We're doing this");
+        if(people.size() > 1 || people.size() == 0){
+            System.out.println("Couldnt find name in database");
+            return "Undefined";
```

Comment:
I am not sure the service should be the one to replace this with a placeholder, that logic might better live in the email service and let this return null.